File: Ejemplo07/app/src/main/java/com/naldana/pokedesk/MainActivity.java
package com.naldana.pokedesk;

import android.os.AsyncTask;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import com.naldana.pokedesk.utilities.NetworkUtils;

import java.io.IOException;
import java.net.URL;

public class MainActivity extends AppCompatActivity {

    EditText mPokemonNumber;
    Button mSearchButton;
    TextView mResultText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        bindView();

        mSearchButton.setOnClickListener(view -> {
            String pokemonNumber = mPokemonNumber.getText().toString().trim();
            if (pokemonNumber.isEmpty()) {
                mResultText.setText(R.string.text_nothing_to_show);
            } else {

                new FetchPokemonTask().execute(pokemonNumber);
            }
        });
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
    }

    void bindView() {
        mPokemonNumber = findViewById(R.id.et_pokemon_number);
        mSearchButton = findViewById(R.id.bt_search_pokemon);
        mResultText = findViewById(R.id.tv_result);
    }


    private class FetchPokemonTask extends AsyncTask<String, Void, String> {

        @Override
        protected String doInBackground(String... pokemonNumbers) {

            if (pokemonNumbers.length == 0) {
                return null;
            }

            String ID = pokemonNumbers[0];

            URL pokeAPI = NetworkUtils.buildUrl(ID);

            try {
                String result = NetworkUtils.getResponseFromHttpUrl(pokeAPI);
                return result;
            } catch (IOException e) {
                e.printStackTrace();
                return "";
            }
        }

        @Override
        protected void onPostExecute(String pokemonInfo) {
            if (pokemonInfo != null || !pokemonInfo.equals("")) {
                mResultText.setText(pokemonInfo);
            } else {
                mResultText.setText(getString(R.string.text_not_found_message));
            }
        }
    }
}

File: Ejemplo06/app/src/main/java/com/naldana/ejemplo06/MainActivity.kt
package com.naldana.ejemplo06

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private var insideCounter = 0
    private var outsideCounter = 0
    private val insideCounterKey = "insideCounter"
    private val outsideCounterKey = "outsideCounter"


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        if (savedInstanceState != null) {
            restoreCounters(savedInstanceState)
        }

        addEventListener()

    }

    private fun restoreCounters(savedInstanceState: Bundle) {

        insideCounter = savedInstanceState.getInt(insideCounterKey)
        outsideCounter = savedInstanceState.getInt(outsideCounterKey)

        displayInside.text = insideCounter.toString()
        displayOutside.text = outsideCounter.toString()
    }

    private fun addEventListener() {

        addInside.setOnClickListener {
            insideCounter++
            displayInside.text = insideCounter.toString()
        }

        addOutside.setOnClickListener {
            outsideCounter++
            displayOutside.text = outsideCounter.toString()
        }
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)

        outState.putInt(insideCounterKey, insideCounter)
        outState.putInt(outsideCounterKey, outsideCounter)
    }
}